src/stacked_borrows/stack.rs

```diff
@@ -11,14 +11,80 @@ use std::ops::Range;
 #[cfg(feature = "stack-cache")]
 const CACHE_LEN: usize = 32;
 
+// We store the permission in the low bits of the tag, leaving only 62 bits for the tag. But that
+// should be more than enough.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+struct PackedItem(u64);
+
+impl PackedItem {
+    const TAG_MASK: u64 = u64::MAX >> 3;
+    const PERM_MASK: u64 = 0x3 << 61;
+    const PROTECTED_MASK: u64 = 0x1 << 63;
```

Comment:
3, 61, 63... lots of magic constants. I guess there is a pattern here?^^

Reply:
These basically form bitfields, but I don't know a good way to express this in Rust. I suppose I could put in a const assertion that these masks add up to `u64::MAX`?

Reply:
Oh I see, you want to say "bits N..M are set". Please at least add comments with the respective N and M.